src/test/java/de/bonndan/nivio/model/GroupTest.java

```diff
@@ -0,0 +1,42 @@
+package de.bonndan.nivio.model;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class GroupTest {
+
+    @Test
+    void doesNotAllowEmptyIdentifier() {
+        assertThrows(Exception.class, () -> new Group(""));
+    }
+
+    @Test
+    void getItemsIsImmutable() {
+        Group g = new Group("foo");
+        assertThrows(Exception.class, () -> g.getItems().add(new Item("a", "b")));
```

Comment:
I am a bit surprised that this throws an exception.

I would have expected that the returned `Set` (but not the original one) is modified.

Reply:
The point is that group "foo" would not allow that an item of group "a" can be added (item.group cannot be changed anymore). 

Messing around with strings should be replaced in #212

Reply:
I was mainly surprised by the `Set.copyOf()` behavior.